pkg/util/time.go

```diff
@@ -0,0 +1,23 @@
+package util
+
+import (
+	"math"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/weaveworks/common/httpgrpc"
+)
+
+// ParseTime parses the string into an int64, milliseconds since epoch.
+func ParseTime(s string) (int64, error) {
```

Comment:
This code seems be copied from `pkg/querier/queryrange/query_range.go`. Deduplicating would be good.